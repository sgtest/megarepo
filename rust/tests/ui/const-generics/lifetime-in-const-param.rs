// https://github.com/rust-lang/rust/issues/113462

struct S2<'b>(&'b ());

struct S<'a, const N: S2>(&'a ());
//~^ ERROR missing lifetime specifier [E0106]
//~| ERROR `S2<'_>` is forbidden as the type of a const generic parameter

fn main() {}
