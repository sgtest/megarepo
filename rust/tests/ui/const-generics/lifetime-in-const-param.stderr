error[E0106]: missing lifetime specifier
  --> $DIR/lifetime-in-const-param.rs:5:23
   |
LL | struct S<'a, const N: S2>(&'a ());
   |                       ^^ expected named lifetime parameter

error: `S2<'_>` is forbidden as the type of a const generic parameter
  --> $DIR/lifetime-in-const-param.rs:5:23
   |
LL | struct S<'a, const N: S2>(&'a ());
   |                       ^^
   |
   = note: the only supported types are integers, `bool` and `char`
   = help: add `#![feature(adt_const_params)]` to the crate attributes to enable more complex and user defined types

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0106`.
